site/src/pages/configuration.mdx

```diff
@@ -132,6 +132,97 @@ Packages map 1:1 with npm packages. Along with specifying the `entrypoints` opti
 }
 ```
 
+### `exports` (experimental)
+
+```ts
+| boolean
+| {
+    envConditions?: ("browser" | "worker")[];
+    extra?: Record<string, JSONValue>;
+  };
+```
+
+The `exports` config allows you to opt-in to generating an `exports` field.
+
+Using the `exports` field enables a couple of things:
+
+- Importing non-root entrypoints in Node.js ESM
+- Disallowing importing modules that aren't specified in the `exports` field
+- More specific builds for certain environments
+
+> Note that Preconstruct's support for the `exports` field does not currently include generating ESM compatible with Node.js. While ESM builds are generated, they are targeting bundlers, not Node.js or browsers directly so they use the `module` condition, not the `import` condition.
+
+Note that adding an `exports` field can arguably be a breaking change, you may want to use the `extra` option to add more exports so that imports that worked previously still work or only add the `exports` field in a major version.
+
+To opt into this experimental feature, you must enable it in the root of your project by setting the `exports` experimental flag in your preconstruct config section of your `package.json` file to `true`.
+
+```diff
+{
+  "name": "@sample/repo",
+  "version": "0.0.0",
+  "preconstruct": {
++    "___experimentalFlags_WILL_CHANGE_IN_PATCH": {
++      "exports": true
++    },
+  }
+}
+```
+
+Additionally, you'll also need to enable the feature on each individual pacakge.
```

Comment:
```suggestion
Additionally, you'll also need to enable the feature on each individual package.
```

Kinda a bummer - but I guess it's better for bigger monorepos that may not want to add this field everywhere just yet. 

It would be cool to have a way to enforce that every package specifies this config, without it I need to be on my toes when adding new packages (arguably I usually copy-paste this stuff but the risk is still there).

Reply:
Yeah, I was always planning to have project-level config for it since that's what you'll probably want to do but didn't add it to this PR, might just add that now